.github/workflows/testing.yml

```
@@ -0,0 +1,87 @@
+ name: Continuous Integration
+ 
+ on:
+   push:
+     branches:
+     - '*'
+   pull_request:
+     branches:
+     - '*'
+   schedule:
+     - cron: '59 23 * * *'
+   workflow_dispatch:
+    inputs:
+      version:
+        description: Manual CI Run
+        default: test
+        required: false
+
+ jobs:
+   tests:
+     name: ${{ matrix.os }}, ${{ matrix.environment-file }}
+     runs-on: ${{ matrix.os }}
+     timeout-minutes: 30
+     strategy:
+       matrix:
+         os: [ubuntu-latest]
+         environment-file: [
+            ci/310-oldest.yaml,
+            ci/310-latest.yaml,
+            ci/311-latest.yaml,
+            ci/312-latest.yaml,
+            ci/312-dev.yaml,
+         ]
+         include:
+           - environment-file: ci/312-latest.yaml
+             os: macos-latest
+           - environment-file: ci/312-latest.yaml
+             os: macos-14 # Apple Silicon
```

```suggestion
             os: macos-latest # Apple Silicon
           - environment-file: ci/312-latest.yaml
             os: macos-13 # Intel
```

GHA default is being migrated.

---

@martinfleis Is [this](https://github.blog/changelog/2024-04-01-macos-14-sonoma-is-generally-available-and-the-latest-macos-runner-image/) the news?